app/assets/javascripts/discourse/app/initializers/clean-dom-on-route-change.js

```diff
@@ -26,7 +26,10 @@ function _clean(transition) {
     activeElement.blur();
   }
 
-  this.lookup("route:application").send("closeModal");
+  let route = this.lookup("route:application");
+  if (!route.isDestroying && !route.isDestroyed) {
+    route.send("closeModal");
```

Comment:
Test failures look related to this change. I'm not sure, haven't run this branch locally, but many of the tests failing have interactions with the modal.

Reply:
I don't think that's the case since we start getting failures in the commits before that change. Also, that change became necessary because actions were taking place on a destroyed/destroying route, which causes its own issues.

Reply:
I should correct myself and say that it seems like it could be related, but that this change has become necessary for the above reason.

Reply:
@pmusaraj any suggestions on how to deal with this?

Reply:
Not sure, let me pull your PR locally and see if I can repro the failing tests.

Reply:
I suspect you'll get at least some tabindex failures in each run, but not always the same one. I find that when I run them solo they generally pass.

Reply:
I may have found something useful. 

On your branch, I reverted up to the first bad commit and the first test error I see there has this: 

```
Attempted to call .send() with the action '_collectTitleTokens' on the destroyed route 'adminBadges'.
```

That seems to have something to do with the `destroy` action in the `admin-badges/show` controller. It fails pretty quickly as well with a test run filtered to "Admin - Badges".

Reply:
Renaming that action to `destroyBadges` fixes the "Admin - Badges" tests and then sorme I get pretty much a clean full run locally (minus one test that fails on `main` too). Not sure why this name collision didn't cause an issue before... but hopefully that's all that is needed. 

The diff is 

```
diff --git a/app/assets/javascripts/admin/addon/controllers/admin-badges/show.js b/app/assets/javascripts/admin/addon/controllers/admin-badges/show.js
index 1e06aea265..cc33f5178a 100644
--- a/app/assets/javascripts/admin/addon/controllers/admin-badges/show.js
+++ b/app/assets/javascripts/admin/addon/controllers/admin-badges/show.js
@@ -207,7 +207,7 @@ export default class AdminBadgesShowController extends Controller.extend(
   }

   @action
-  destroy() {
+  destroyBadge() {
     const adminBadges = this.adminBadges.model;
     const model = this.model;

diff --git a/app/assets/javascripts/admin/addon/templates/admin-badges/show.hbs b/app/assets/javascripts/admin/addon/templates/admin-badges/show.hbs
index 520790be3b..1cf2bbab5a 100644
--- a/app/assets/javascripts/admin/addon/templates/admin-badges/show.hbs
+++ b/app/assets/javascripts/admin/addon/templates/admin-badges/show.hbs
@@ -159,7 +159,7 @@
       <DButton @class="btn-primary" @action={{action "save"}} @type="submit" @disabled={{this.saving}} @label="admin.badges.save" />
       <span class="saving">{{this.savingStatus}}</span>
       {{#unless this.readOnly}}
-        <DButton @action={{action "destroy"}} @class="btn-danger" @label="admin.badges.delete" />
+        <DButton @action={{action "destroyBadge"}} @class="btn-danger" @label="admin.badges.delete" />
       {{/unless}}
     </div>
   </form>
```

Reply:
@pmusaraj that's the purpose of that last commit that you raised questions about, to check that we don't call things against a destroyed route.

Reply:
@wagenet not sure whether you saw this [comment](https://github.com/discourse/discourse/pull/17672#discussion_r932728541)... 

In commit [2f24d33](https://github.com/discourse/discourse/pull/17672/commits/2f24d33f10d694e5766b5882a22db0511ac07271), for example, this test fails and it invokes the `destroy` action even though nothing in the test should trigger that specific action. A quick `console.trace` shows indeed that there is an Ember internal `destroy` function that is called.

Reply:
@pmusaraj ah! Somehow your second comment hadn't loaded. There are subtle differences between how inheritance and overriding works between `EmberObject.extend` and the native `class extends`. It seems likely that the issue here could be explained by that. I'll push out a fix tomorrow.